HeterogeneousCore/AlpakaInterface/interface/radixSort.h

```diff
@@ -0,0 +1,250 @@
+    //move negative first...
+
+    auto& firstNeg = alpaka::declareSharedVar<uint32_t, __COUNTER__>(acc);
+    firstNeg = a[ind[0]] < 0 ? 0 : size;
+    alpaka::syncBlockThreads(acc);
+
+    // find first negative
+    for_each_element_in_block_strided(acc, size - 1, [&](uint32_t idx) {
+      if ((a[ind[idx]] ^ a[ind[idx + 1]]) < 0)
+        firstNeg = idx + 1;
+    });
+    alpaka::syncBlockThreads(acc);
+
+    for_each_element_in_block_strided(acc, size, firstNeg, [&](uint32_t idx) { ind2[size - idx - 1] = ind[idx]; });
+    alpaka::syncBlockThreads(acc);
+
+    for_each_element_in_block_strided(acc, firstNeg, [&](uint32_t idx) { ind2[idx + size - firstNeg] = ind[idx]; });
+    alpaka::syncBlockThreads(acc);
+
+    for_each_element_in_block_strided(acc, size, [&](uint32_t idx) { ind[idx] = ind2[idx]; });
+  }
+
+  template <typename TAcc,
+            typename T,  // shall be interger
+            int NS,      // number of significant bytes to use in sorting
+            typename RF>
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE void radixSortImpl(
+      const TAcc& acc, T const* __restrict__ a, uint16_t* ind, uint16_t* ind2, uint32_t size, RF reorder) {
+#if (defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && defined(__CUDA_ARCH__)) || \
+    (defined(ALPAKA_ACC_GPU_HIP_ENABLED) && defined(__HIP_DEVICE_COMPILE__))
+    const uint32_t threadIdxLocal(alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[0u]);
+    const uint32_t blockDimension(alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[0u]);
+
+    constexpr int d = 8, w = 8 * sizeof(T);
+    constexpr int sb = 1 << d;
+    constexpr int ps = int(sizeof(T)) - NS;
+
+    auto& c = alpaka::declareSharedVar<int32_t[sb], __COUNTER__>(acc);
+    auto& ct = alpaka::declareSharedVar<int32_t[sb], __COUNTER__>(acc);
+    auto& cu = alpaka::declareSharedVar<int32_t[sb], __COUNTER__>(acc);
+    auto& ibs = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+    auto& p = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+
+    ALPAKA_ASSERT_OFFLOAD(size > 0);
+    ALPAKA_ASSERT_OFFLOAD(blockDimension >= sb);
+
+    p = ps;
+
+    auto j = ind;
+    auto k = ind2;
+
+    for_each_element_in_block_strided(acc, size, [&](uint32_t idx) { j[idx] = idx; });
+    alpaka::syncBlockThreads(acc);
+
+    while (alpaka::syncBlockThreadsPredicate<alpaka::BlockAnd>(acc, (p < w / d))) {
+      for_each_element_in_block_strided(acc, sb, [&](uint32_t idx) { c[idx] = 0; });
+      alpaka::syncBlockThreads(acc);
+
+      // fill bins
+      for_each_element_in_block_strided(acc, size, [&](uint32_t idx) {
+        auto bin = (a[j[idx]] >> d * p) & (sb - 1);
+        alpaka::atomicAdd(acc, &c[bin], 1, alpaka::hierarchy::Threads{});
+      });
+      alpaka::syncBlockThreads(acc);
+
+      // prefix scan "optimized"???...
+      for_each_element_in_block(acc, sb, [&](uint32_t idx) {
+        auto x = c[idx];
+        auto laneId = idx & 0x1f;
+
+        for (int offset = 1; offset < 32; offset <<= 1) {
+#if defined(__CUDA_ARCH__)
```

Comment:
can you add a `// TODO` here, to use the alpaka equivalent as soon as it's available ?
(see https://github.com/alpaka-group/alpaka/pull/1924)